### PR TITLE
UCT/IB: post recv fixes

### PR DESCRIPTION
**Pull request by alex-mikheev**
reduce batch size to 16 from 32
{rc,ud}_verbsa: always try to post recv wqe
ud: do not allow setting tx_batch size less then ud batch size
@yosefe

### CODE REVIEW

File: src/uct/ib/ud/verbs/ud_verbs.c [hunk 1]
```diff
@@ -472,6 +476,12 @@ static UCS_CLASS_INIT_FUNC(uct_ud_verbs_iface_t, uct_pd_h pd, uct_worker_h worke
     self->super.ops.async_progress = uct_ud_verbs_iface_async_progress;
     self->super.ops.tx_skb         = uct_ud_verbs_ep_tx_ctl_skb;
 
+    if (self->super.config.rx_max_batch < UCT_IB_MAX_WC) {
+        ucs_warn("max batch is too low (%d < %d), performance may be impacted",
```
Comment (yosefe) [inline src/uct/ib/ud/verbs/ud_verbs.c]: this will be printed every time, right?
can we print it during iint only?

Comment (yosefe) [inline src/uct/ib/ud/verbs/ud_verbs.c]: ok, didn't read the diff right.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1107/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1109/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/532/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/534/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
autobot build

**Comment by shamisp** [comment]
Works on our setup

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/541/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1119/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/542/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1120/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1121/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/543/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
@yosefe can you take a look and merge ? 10x

File: src/ucs/datastruct/arbiter.c [hunk 2]
```diff
@@ -22,12 +22,10 @@ void ucs_arbiter_group_init(ucs_arbiter_group_t *group)
 
 void ucs_arbiter_cleanup(ucs_arbiter_t *arbiter)
 {
-    ucs_assert(arbiter->current == NULL);
```
Comment (yosefe) [inline src/ucs/datastruct/arbiter.c]: why did you remove these assertions?

File: src/uct/ib/ud/base/ud_ep.c [hunk 3]
```diff
@@ -83,7 +83,6 @@ static UCS_CLASS_CLEANUP_FUNC(uct_ud_ep_t)
     uct_ud_iface_remove_ep(iface, self);
     uct_ud_iface_cep_remove(self);
    /* TODO: in disconnect ucs_frag_list_cleanup(&self->rx.ooo_pkts); */
-    ucs_assert(self->tx.pending.ops == UCT_UD_EP_OP_NONE);
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_ep.c]: maybe need to purge the group here? i don't feel comfortable keeping elements in a deleted group, may lead to use-after-free

File: src/ucs/datastruct/arbiter.c (hunk 2 above)
Comment (alex-mikheev) [inline src/ucs/datastruct/arbiter.c]: because we can not guarantee that there will be no pending op during interface cleanup.

Comment (yosefe) [inline src/ucs/datastruct/arbiter.c]: i think we have to actually clean the arbiter before destroying it

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1122/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-sharp-pr/2/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/545/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/546/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
@yosefe hopefully this will be the last last look :)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1130/ for details.

**Comment by yosefe** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1149/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1151/ for details.

**Pull request merged**